sdk/storage/Azure.Storage.Blobs/src/BlobStorageContext.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Pipeline;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Shared;
+
+namespace Azure.Storage.Blobs
+{
+    /// <summary>
+    /// Contains common properties used to create clients.
+    /// </summary>
+    internal class BlobStorageContext : StorageContext
+    {
+        public virtual BlobClientOptions.ServiceVersion Version { get; private set; }
+
+        public virtual CustomerProvidedKey? CustomerProvidedKey { get; private set; }
+
+        public virtual string EncryptionScope { get; private set; }
```

Comment:
Note that I didn't include `ClientSideEncryptionOptions` or `Uri` here.  This is because `ClientSideEncryptionOptions` is only a property on `BlobClient` and `BlobBaseClient`.  I didn't include `Uri` because we have lots of methods that just transform the Uri and return a new client, like `WithVersion()` and `WithSnapshot()`